### PR TITLE
[DO NOT MERGE] - Apriltag3

### PR DESCRIPTION
**Pull request by afdaniele**
## Description

- Removed old AprilTag packages v1 and v2
- Removed copy of C AprilTag library (now it is pre-built on PyPi)
- Added only one AprilTag package `apriltag` containing a Python-based detector node and a post-processing node
- Detector node is fully compliant with DTROS

**[DO NOT MERGE]**
This PR cannot be merge in the first wave of changes to daffy-new-deal. The first wave of changes has the objective of getting the IN demo back to daffy. Though this code is based on a new version of DTROS that is not backward compatible. The second wave of changes will include the new version of DTROS (PR: https://github.com/duckietown/dt-ros-commons/pull/28), which will clear the way for this PR.
**[DO NOT MERGE]**

## Type of change

Please mark the relevant options with X:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change is constrained only to one package, any interfaces to nodes from other packages are maintained
- [ ] This change is constrained only to packages in this repository, any interfaces to nodes from other repositories are maintained
- [ ] This change required a documentation update
- [ ] This change is only a documentation update
- [ ] This change restructures the repository (at level higher than package level)
- [ ] This change requires a change in the Duckiebook (if yes, please describe and, if possible, make a pull request for an edit in the book and link it here)

## Tests

Please describe all tests ran and their outcome. It is highly recommended that you at least do the tests listed by default, even if you only changed the documentation.

- [ ] Made the Duckiebot move with keyboard control
- [X] Visually checked the published camera stream
- [ ] Verified that the lane following demo works
- [ ] Verified that the indefinite navigation demo works

## Style compliance
Please make sure that all your changes conform to the Duckietown (code style)[link] and code (documentation style)[link] guidelines:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I checked that this pull request is compliant with the Duckietown code style
- [X] I checked that this pull request is compliant with the Duckietown code documentation style

## Next steps
Please describe if you have any questions, comments, or doubts towards the Duckietown software guardians regarding implementation, code style, running tests, documentation, etc.

## FOR THE REVIEWERS
__IMPORTANT:__ DO NOT CHANGE THIS SECTION!

- [ ] Functionality (is the change necessary and is it implemented in the most appropriate way): @liampaull or @afdaniele
- [ ] Integration testing (verify the tests and run additional if necessary): @?
- [ ] Code style compliance: @gibernas
- [ ] Documentation compliance: @aleksandarpetrov

_To the reviewers:_ If you deem necessary, request code edits, more tests, documentation update, or additional reviewing. __Do not review your own code!__

_PR Template, last edit 23 Aug 2019_

### CODE REVIEW

**Comment by liampaull** [comment]
linked to a PR in ros-commons.

**Comment by afdaniele** [comment]
@AleksandarPetrov can you please review?

File: packages/apriltag/src/apriltag_detector_node.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+
+class AprilTagDetector(DTROS):
+
+    def __init__(self):
+        super(AprilTagDetector, self).__init__(
+            node_name='apriltag_detector_node',
+            node_type=NodeType.PERCEPTION
+        )
+        # get static parameters
```
Comment (AleksandarPetrov) [inline packages/apriltag/src/apriltag_detector_node.py:26]: why are these static? we should be able to change them dynamically

File: packages/apriltag/src/apriltag_detector_node.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            return
+        # ---
+        # turn image message into grayscale image
+        img = self._bridge.imgmsg_to_cv2(data, desired_encoding='mono8')
+        # detect tags
+        tags = self._at_detector.detect(img, True, self._camera_parameters, self.tag_size)
+        # draw the detections on an image (if needed)
+        if self._img_pub.anybody_listening() and not self._img_pub_busy:
+            self._img_pub_busy = True
+            Thread(target=self._publish_detections_image, args=(img, tags)).start()
```
Comment (AleksandarPetrov) [inline packages/apriltag/src/apriltag_detector_node.py:88]: Why a separate thread? Isn't this dangerous if generating the image is too slow and we start threads faster than they exit?

File: packages/apriltag/src/apriltag_detector_node.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                fontScale=0.8,
+                color=(0, 0, 255)
+            )
+        # pack image into a message
+        img_msg = self._bridge.cv2_to_compressed_imgmsg(color_img)
+        # ---
+        self._img_pub.publish(img_msg)
+        self._img_pub_busy = False
+
+    def _cinfo_cb(self, data):
```
Comment (AleksandarPetrov) [inline packages/apriltag/src/apriltag_detector_node.py:165]: we should not shutdown the subscriber. If the camera resolution is changed, we need to update the camera parameters and re-initalize the Detector

File: packages/apriltag/src/apriltag_postprocessing_node.py [hunk 4]
```diff
@@ -12,22 +16,21 @@ class AprilPostPros(object):
     """ """
     def __init__(self):
```
Comment (AleksandarPetrov) [inline packages/apriltag/src/apriltag_postprocessing_node.py:17]: This is not DTROS-ified yet...

Comment (afdaniele) [reply packages/apriltag/src/apriltag_postprocessing_node.py:17]: DTROS-ification is not part of this batch of changes

**Pull request merged**